### PR TITLE
Park drive animation

### LINKED ISSUE #65: Making the transition between park and drive more responsive and alive
As a driver of my luxorious AlSet, I would like the interface to represent my expected level of polish and luxury when it comes to driving in order to make me feel like I'm not only driving a luxurious product, I am also using one.

Acceptance criteria:
- [x] Animate the transition between Park and Drive
- [x] Animate the transitions between Drive and Park

### PR DESCRIPTION
**Pull request by ktpa**
Leaving betong mqtt up until merged for testing purposes. Works fine on emulator and on my phone. Try it out, it's pretty sweet.

### CODE REVIEW

File: client/app/src/main/java/group01/smartcar/client/view/ProximityIndicator.java [hunk 1]
```diff
@@ -52,23 +52,23 @@ public ProximityIndicator(Context context) {
 
         getHolder().addCallback(this);
         getHolder().setFormat(PixelFormat.TRANSLUCENT);
-        setZOrderOnTop(true);
+        //setZOrderOnTop(true);
```
Comment (JensTheS) [inline client/app/src/main/java/group01/smartcar/client/view/ProximityIndicator.java]: Clean up commented code that is no longer used?
Same for line 63, 71

**Review by JensTheS — COMMENTED** [review]
A suitable metaphor: "Spring adds new life and new beauty to all that is." 
The animation you coded is Spring at its best!

File: client/app/src/main/java/group01/smartcar/client/activity/DrivingActivity.java [hunk 2]
```diff
@@ -77,12 +77,15 @@
...
     private Speedometer speedometer;
     private ProximityIndicator proximityIndicator;
+    private ImageView speedometerOverlay;
     private Vibrator vibrator;
     private ImageView micButton;
     private SpeechListener speechListener;
     private ImageView batteryImage;
     private ImageView backButton;
 
+    private int shortAnimationDuration;
```
Comment (dekmm) [inline client/app/src/main/java/group01/smartcar/client/activity/DrivingActivity.java]: Consider making this a constant:
```
private static final int SHORT_ANIMATION_DURATION = android.R.integer.config_shortAnimTime;
```

**Review by JensTheS — APPROVED** [review]
Looks great!

**Pull request merged**